src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java

```diff
@@ -0,0 +1,31 @@
+package de.jcm.discordgamesdk.impl.events;
+
+import de.jcm.discordgamesdk.Core;
+import de.jcm.discordgamesdk.LogLevel;
+import de.jcm.discordgamesdk.impl.Command;
+import de.jcm.discordgamesdk.impl.EventHandler;
+import de.jcm.discordgamesdk.user.DiscordUser;
+
+public class ActivityJoinEvent {
+    public static class Handler extends EventHandler<DiscordUser>
+    {
+        public Handler(Core.CorePrivate core)
+        {
+            super(core);
+        }
+
+        @Override
+        public void handle(Command command, DiscordUser user)
+        {
+            core.log(LogLevel.VERBOSE, "Secret: " + command.getData().getAsJsonObject().get("secret").getAsString());
+            core.getEventAdapter().onActivityJoin(command.getData().getAsJsonObject().get("secret").getAsString());
```

Comment:
I would personally not get the secret directly from the JSON object, but instead use the existing data mechanism.

e.g.:
```java
public static class Data {
    String secret;
}

public static class Handler extends EventHandler<Data> {

    @Override
    public void handle(Command command, Data data) {
        // use data.token
    }

    @Override
    public Class<?> getDataClass() {
        return Data.class;
    }
}